edweiss-app/components/courses/MaterialComponent.tsx

```diff
@@ -177,30 +265,191 @@ const MaterialComponent: ReactComponent<MaterialProps> = ({ mode, onSubmit, onDe
         }
     };
 
+
+
+    //===============================================================
+    //======================== File Upload ==========================
+    //===============================================================
+
+    // Handle the file picker
+    const handlePickFile = async () => {
+        try {
```

Comment:
You define a lot of functions inside your Material component. I see you still got sufficient coverage (excellent job on that), but if you want to increase it further, consider separating the functions from the component, to be able to test them easily.

Reply:
Yes this is a good idea! I will do it if possible (depends if I have time to do it) in a future PR.